csharp_dotnetcore/11.QnAMaker/Startup.cs

```diff
@@ -0,0 +1,164 @@
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        /// <summary>
+        /// Gets the configuration that represents a set of key/value application configuration properties.
+        /// </summary>
+        /// <value>
+        /// The IConfiguration that represents a set of key/value application configuration properties.
+        /// </value>
+        public IConfiguration Configuration { get; }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// </summary>
+        /// <param name="services">Specifies the contract for a collection of service descriptors.</param>
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Load the connected services from .bot file
+            // BUGBUG: This needs to change to common bot file, once emulator understand appInsights type.
+            var botConfig = BotConfiguration.LoadAsync(@".\QnaBot.bot").Result;
+
+            // Initialize Bot Connected Services Clients
+            var connectedServices = InitBotServices(botConfig);
+            services.AddSingleton(sp => connectedServices);
+
+            services.AddSingleton(sp => botConfig);
+
+            services.AddBot<QnABot>(options =>
+            {
+                options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+
+                // Add MyAppInsightsLoggerMiddleware (logs activity messages into Application Insights)
+                var appInsightsLogger = new MyAppInsightsLoggerMiddleware(connectedServices.TelemetryClient.InstrumentationKey, logUserName: true, logOriginalMessage: true);
+                options.Middleware.Add(appInsightsLogger);
+            });
+        }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        /// </summary>
+        /// <param name="app">The application builder.  This provides the mechanisms to configure an application's request pipeline.</param>
+        /// <param name="env">Provides information about the web hosting environment an application is running in.</param>
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseDefaultFiles()
+                .UseStaticFiles()
+                .UseBotFramework();
+        }
+
+        /// <summary>
+        /// Initialize the bot's references to external services.
+        ///
+        /// For example, Application Insights and QnaMaker services
+        /// are created here.  These external services are configured
+        /// using the BotConfigure class (based on the contents of your ".bot" file).
+        /// </summary>
```

Comment:
Consider adding a comment here around object lifecycle, and how these are singletons and the IBot class is created new for each activity.